pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -147,14 +135,41 @@ func endExecutionSpan(processTrace func(p *api.Payload), requestID string, endTi
 	})
 }
 
-func convertRawPayload(rawPayload string) invocationPayload {
-	//Need to remove unwanted text from the initial payload
-	reg := regexp.MustCompile(`{(?:|(.*))*}`)
-	subString := reg.FindString(rawPayload)
+// parseLambdaFunction removes extra data sent by the proxy that surrounds
+// a JSON payload. For example, for `a5a{"event":"aws_lambda"...}0` it would remove
+// a5a at the front and 0 at the end, and just leave a correct JSON payload.
+func parseLambdaPayload(rawPayload string) string {
+	leftFound, rightFound := false, false
+	leftIndex, rightIndex := 0, 0
+	for i, char := range rawPayload {
+		if char == '{' {
+			leftIndex = i
+			leftFound = true
+			break
+		}
+	}
+	for i := range rawPayload {
```

Comment:
https://pkg.go.dev/strings#LastIndex ?

Reply:
Awesome, should have dug a little deeper. Fixed!